SHA image tags cannot be converted to tags

I get this error for the images which use the sha as tag. 
The velero helm chart was configured per default to use it. 
I've changed it to a normal version tag but since this kind of sha notation seems to be valid it would be good if that's also supported.

```
INFO[0004] Fetching containers for namespace             namespace=velero
DEBU[0004] Fetched containers in namespace               images="map[velero/velero-plugin-for-aws:v1.0.0:true velero/velero@sha256:0c74f1d552ef25a4227e582f4c0e6b3db3402abe196595ee9442ceeb43b99696:true]" namespace=velero
panic: interface conversion: reference.canonicalReference is not reference.NamedTagged: missing method Tag

goroutine 1 [running]:
github.com/arminc/k8s-platform-lcm/internal/kubernetes.ImageStringToContainerStruct(0xc0000a2f00, 0x55, 0x0, 0x0, 0x0,
0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /home/travis/gopath/src/github.com/arminc/k8s-platform-lcm/internal/kubernetes/images.go:17 +0x1cf
github.com/arminc/k8s-platform-lcm/internal/kubernetes.GetContainersFromNamespaces(0xc000230f20, 0x16, 0x16, 0x1b47300, 0xc00004fe60, 0xc000686e08, 0x4fa815)
        /home/travis/gopath/src/github.com/arminc/k8s-platform-lcm/internal/kubernetes/k8s.go:37 +0x2e6
github.com/arminc/k8s-platform-lcm/internal.Execute(0x0, 0x7ffc146727f3, 0x13, 0x1, 0x0, 0x0, 0x100, 0x0, 0x0, 0x0, ...)
        /home/travis/gopath/src/github.com/arminc/k8s-platform-lcm/internal/lcm.go:39 +0x800
main.main()
        /home/travis/gopath/src/github.com/arminc/k8s-platform-lcm/cmd/lcm/main.go:59 +0x20c
```

--- a/pkg/kubernetes/images.go
+++ b/pkg/kubernetes/images.go
@@ -23,15 +23,18 @@ func (c Image) WithoutLibrary() string {
 // ImagePathToImage converts image string to container information
 // In case of an error it returns an empty Image
 func ImagePathToImage(imagePath string) (Image, error) {
+	version := "0"
 	image, err := reference.ParseNormalizedNamed(imagePath)
 	if err != nil {
 		return Image{}, errors.Wrap(err, "Failed to pars image name")
 	}
 	image = reference.TagNameOnly(image) // adds tag latest if no tag is set
 
-	version := image.(reference.NamedTagged).Tag()
-	if version == "latest" {
-		version = "0" // tag 'latest' can't be compared
+	if !strings.Contains(imagePath, "@sha256:") {
+		version = image.(reference.NamedTagged).Tag()
+		if version == "latest" {
+			version = "0" // tag 'latest' can't be compared
+		}
 	}
 
 	return Image{

--- a/pkg/kubernetes/images_test.go
+++ b/pkg/kubernetes/images_test.go
@@ -53,3 +53,9 @@ func TestPodStringToPodStructWithPort(t *testing.T) {
 	expected := container("gcr.io:443/somebody/test:1.3", "gcr.io:443", "somebody/test", "1.3")
 	assert.Equal(t, expected, pod, "With port")
 }
+
+func TestPodStringToPodStructWithSha(t *testing.T) {
+	pod, _ := ImagePathToImage("gcr.io:443/somebody/test@sha256:6f82e2b0464f50da8104acd7363fb9b995001ddff77d248379f8788e78946143")
+	expected := container("gcr.io:443/somebody/test@sha256:6f82e2b0464f50da8104acd7363fb9b995001ddff77d248379f8788e78946143", "gcr.io:443", "somebody/test", "0")
+	assert.Equal(t, expected, pod, "With sha")
+}

